phpBB/includes/diff/engine.php

```diff
@@ -285,7 +285,7 @@ function _diag($xoff, $xlim, $yoff, $ylim, $nchunks)
 				$matches = $ymatches[$line];
 
 				reset($matches);
-				while (list(, $y) = each($matches))
+				while ($y = current($matches))
```

Comment:
That's not the equivalent. `current()` will not move the pointer and therefore will always return the first element in this case.

Reply:
I think a simple `foreach ($matches as $y)` would be the correct replacement.

Reply:
The issue with `foreach` is that this 1st `while` loop is to be broken on `empty($this->in_seq[$y])` and the 2nd `while` loop (right below) starts with no pointer reset from the current array position.
I wonder what's gonna happen if we use `foreach` for both.

Reply:
The logic should be restored now.